devops/providers/azure-devops/templates/infrastructure/cicd/azure-pipelines.yaml

```diff
@@ -0,0 +1,125 @@
+variables:
+
+# location of files used by pipeline
+- name: PIPELINE_SCRIPTS_ROOT
+  value: cicd
+- name: BUILD_DIR
+  value: build
+
+# dependency versions
+- name: GO_VERSION
+  value: '1.12.5'
+- name: TF_VERSION
+  value: '0.12.2'
+
+# terraform flags
+- name: TF_WARN_OUTPUT_ERRORS
+  value: 1
+- name: ARM_PROVIDER_STRICT
+  value: 'true'
+
+# test harness image location and image tagging metadata
+- name: TEST_HARNESS_BASE_IMAGE
+  value: 'msftcse/cobalt-test-base'
+- name: TEST_HARNESS_BUILD_ARTIFACT_IMAGE_PREFIX
+  value: cobalt-build-artifact
+- name: TEST_HARNESS_REPOSITORY_GIT_URL
+  value: https://github.com/microsoft/cobalt.git
+- name: TEST_HARNESS_DIR
+  value: test-harness
+
+
+stages:
+- stage: CI
+  jobs:
+
+  - job: CI
+
+
+    ### BEGIN BUILD TASKS
+
+    pool: $(AGENT_POOL)
+    workspace:
+      clean: all
```

Comment:
This should wipe the workspace and force a clone correct? Why do we need to do this?

Reply:
Yep -- just an extra measure to make sure that we start with a clean slate